ivy/functional/ivy/linear_algebra.py

```diff
@@ -2725,3 +2725,57 @@ def lu_factor(
         A named tuple (LU, pivots).
     """
     return current_backend(A).lu_factor(A, pivot=pivot, out=out)
+
+
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+@handle_exceptions
+@handle_array_like_without_promotion
+def norm(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    ord: Union[int, float, str] = 2,
+    axis: Optional[Union[int, Tuple[int, ...]]] = None,
+    keepdims: bool = False,
+    out: Optional[Union[ivy.Array, ivy.NativeArray]] = None,
+) -> Union[ivy.Array, ivy.NativeArray]:
+    """
+    Parameters
+    ----------
+    x
+        tensor of shape (*, m, n) where * is zero or more batch dimensions.
+
+    ord
+        Order of the norm. Supported values are 'fro', 'nuc', 'inf', -inf, 1, -1, 2, -2,
+        and any other positive integer. Default: 2.
+
+    axis
+        Axis or axes along which to compute the norm. If None, compute the norm over the
+        entire tensor. Default: None.
+
+    keepdims
+        If this is set to True, the axes which are normed over are left in the result as
+        dimensions with size one. With this option, the result will broadcast correctly
+        against the original x. Default: False.
+
+    out
+        tensor to write the output to. Ignored if None. Default: None.
+
+    Returns
+    -------
+    ret
+        tensor of shape (*, m, n) where * is zero or more batch dimensions.
+    """
+    if isinstance(axis, int):
+        return current_backend(x).vector_norm(
+            x, ord=ord, axis=axis, keepdims=keepdims, out=out)
+    elif isinstance(axis, tuple):
+        return current_backend(x).matrix_norm(
+            x, ord=ord, axis=axis, keepdims=keepdims, out=out)
+    elif axis is None and ord is not None:
+        return current_backend(x).vector_norm(
+            x, ord=ord, axis=axis, keepdims=keepdims, out=out)
+    else:
+        raise ValueError("Invalid axis or ord value.")
```

Comment:
Can you please migrate this code to the torch frontend function? avoid making any changes to the ivy API. You can then replace the `current_backend(x).` with just `ivy.`

Reply:
My frontend method:

```python
def norm(input, p=2, dim=None, keepdim=False, *, out=None):
    if isinstance(dim, int):
        return ivy.vector_norm(
            input, ord=ord, axis=dim, keepdims=keepdim, out=out)
    elif isinstance(dim, tuple):
        return ivy.matrix_norm(
            input, ord=ord, axis=dim, keepdims=keepdim, out=out)
    elif dim is None and ord is not None:
        return ivy.vector_norm(
            input, ord=ord, axis=dim, keepdims=keepdim, out=out)
    else:
        raise ValueError("Invalid axis or ord value.")
```



This what is my ivy function returns:
```python
return current_backend(x).norm(x, ord=ord, axis=axis, keepdims=keepdims, out=out)
```
    
And I'm getting this error:
```python
IvyError: numpy: norm: module 'ivy.functional.backends.torch' has no attribute 'norm'
```

Reply:
You don't need a `norm` ivy function at all, like i said there's no need for you to modify anything in the ivy api at all. Notice how you never call an `ivy.norm` in your frontend function

Reply:
Thanks! Can you give me a simple code on how to test this new frontend function? I think I was doing it wrong.